engine/env/spark/src/main/java/org/datacleaner/spark/utils/ResultFilePathUtils.java

```
@@ -19,86 +19,54 @@
+     * in the job properties file with property 'datacleaner.result.hdfs.path'.
+     * The path can be absolute(
      * 'hdfs://[hostname]:[port]/myresults/myjob.analysis.result.dat') or
      * relative('/myresults/myjob.analysis.result.dat'). If no path is set in
      * the properties file and the system is configured to save the results, the
      * default path will be saved to
      * '/datacleaner/results/[jobname]-[timestamp].analysis.result.dat
      * 
-     * @throws URISyntaxException
-     * 
-     * @retur
+     * @return a Resource to use
      */
-    public static String getResultFilePath(final JavaSparkContext sparkContext, final SparkJobContext sparkJobContext)
-            throws URISyntaxException {
-        String resultPath = sparkJobContext.getResultPath();
-        final Configuration hadoopConfiguration = sparkContext.hadoopConfiguration();
-        /**
-         * The default value would be read from hadoop configuration at runtime
-         * from core-site.xml. It represents the machine's hostname and port.
-         * Example: hdfs://bigdatavm:9000
-         **/
-        final String fileSystemPrefix = hadoopConfiguration.get("fs.defaultFS");
-        if (resultPath == null || resultPath.isEmpty()) {
-            resultPath = createPath(fileSystemPrefix, DEFAULT_RESULT_PATH);
+    public static Resource getResultResource(final JavaSparkContext sparkContext,
+            final SparkJobContext sparkJobContext) {
+        final HdfsHelper hdfsHelper = new HdfsHelper(sparkContext);
+
+        final String resultPath;
+        if (Strings.isNullOrEmpty(sparkJobContext.getResultPath())) {
+            resultPath = DEFAULT_RESULT_PATH + '/' + generateResultFilename(sparkJobContext);
         } else {
-            final URI uri = URI.create(resultPath);
-            if (!uri.isAbsolute()) {
-                resultPath = createPath(fileSystemPrefix, resultPath);
+            if (hdfsHelper.isDirectory(sparkJobContext.getResultPath())) {
```

Early validation of the result file path has now been added as of commit  9b731f6 - this is not done in the Main class, but rather in SparkJobContext since this is also where the paths for the job and configuration file is done. And it is done at construction time, which would make it fail fast always.